HoloRepositoryDemoApplication/Assets/HoloRpository/Scripts/StorageConnectionServer.cs

```diff
@@ -0,0 +1,256 @@
+            catch (Exception e)
+            {
+                Debug.LogError($"{e.Message}\n{e.StackTrace}");
+                return;
+            }
+        }
+        #endregion Public Method
+
+        #region Private Common Method
+        private static IEnumerator GetRequest(string uri)
+        {
+            using (UnityWebRequest webRequest = UnityWebRequest.Get(uri))
+            {
+                yield return webRequest.SendWebRequest();
+                WebRequestReturnData = null;
+                if (webRequest.isNetworkError)
+                {
+                    Debug.Log("Web request Error! [Error message]: " + webRequest.error);
+                }
+                else
+                {
+                    WebRequestReturnData = webRequest.downloadHandler.text;
+                }
+            }
+        }
+
+        private static void CopyProperties(object source, object destination)
+        {
+            PropertyInfo[] destinationProperties = destination.GetType().GetProperties();
+            foreach (PropertyInfo destinationPi in destinationProperties)
+            {
+                PropertyInfo sourcePi = source.GetType().GetProperty(destinationPi.Name);
+                destinationPi.SetValue(destination, sourcePi.GetValue(source, null), null);
+            }
+        }
+
+        private static Patient JsonToPatient(JSONNode Json)
+        {
+            Patient patient = new Patient();
+
+            if (Json["pid"].Value == "")
+            {
+                Debug.Log("No response from server with this patient ID!");
+                return patient;
+            }
+
+            try
+            {
+                patient.pid = Json["pid"].Value;
+
+                PersonName name = new PersonName();
+                name.title = Json["name"]["title"].Value;
+                name.full = Json["name"]["full"].Value;
+                name.first = Json["name"]["first"].Value;
+                name.last = Json["name"]["last"].Value;
+                patient.name = name;
+
+                patient.gender = Json["gender"].Value;
+                patient.email = Json["email"].Value;
+                patient.phone = Json["phone"].Value;
+                patient.birthDate = Json["birthDate"].Value;
+                patient.pictureUrl = Json["pictureUrl"].Value;
+
+                Address address = new Address();
+                address.street = Json["address"]["street"].Value;
+                address.city = Json["address"]["city"].Value;
+                address.state = Json["address"]["state"].Value;
+                address.postcode = Json["address"]["postcode"].AsInt;
+                patient.address = address;
+            }
+            catch (Exception e)
+            {
+                Debug.Log("Failed to map patient from response data! \n[Error message]: " + e);
+            }
+                    
+            return patient;
+        }
+
+        private static Hologram JsonToHologram(JSONNode Json)
```

Comment:
hmm I guess mapping manually is one way to do it, but may be a bit cumbersome? You may  want to consider some libraries  that  help with JSON serialisation, see https://stackoverflow.com/questions/2246694/how-to-convert-json-object-to-custom-c-sharp-object

Reply:
Another issue I mentioned, I used LitJson before, it have the  JSON serialisation method, it works on unity but not on UWP(HoloLens), I did a bit research, there are also some people get issues with JSON.NET, bcz both LitJson and JSON.NET are imported by .dll file, and this file is not supported by UWP app, now I'm using SimpleJSON, which is imported by C# scripts, and people say it works well on HoloLens, the problem is it does not have serialisation method, need do it manually